clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/provider/KubernetesV2ArtifactProvider.java

```diff
@@ -19,44 +19,40 @@
 
 import static com.google.common.collect.ImmutableList.toImmutableList;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
-import com.netflix.spinnaker.cats.cache.CacheData;
-import com.netflix.spinnaker.clouddriver.kubernetes.KubernetesCloudProvider;
-import com.netflix.spinnaker.clouddriver.kubernetes.caching.Keys;
 import com.netflix.spinnaker.clouddriver.kubernetes.description.manifest.KubernetesKind;
+import com.netflix.spinnaker.clouddriver.kubernetes.description.manifest.KubernetesManifest;
+import com.netflix.spinnaker.clouddriver.kubernetes.description.manifest.KubernetesManifestAnnotater;
 import com.netflix.spinnaker.kork.artifacts.model.Artifact;
 import java.util.Comparator;
+import java.util.Optional;
+import java.util.stream.Stream;
 import javax.annotation.Nonnull;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class KubernetesV2ArtifactProvider implements ArtifactProvider {
-  private final KubernetesCacheUtils cacheUtils;
-  private final ObjectMapper objectMapper;
+  private final KubernetesAccountResolver accountResolver;
 
   @Autowired
-  KubernetesV2ArtifactProvider(KubernetesCacheUtils cacheUtils, ObjectMapper objectMapper) {
-    this.cacheUtils = cacheUtils;
-    this.objectMapper = objectMapper;
+  KubernetesV2ArtifactProvider(KubernetesAccountResolver accountResolver) {
+    this.accountResolver = accountResolver;
   }
 
   @Override
   public ImmutableList<Artifact> getArtifacts(
       KubernetesKind kind, String name, String location, @Nonnull String account) {
-    String type = String.join("/", KubernetesCloudProvider.ID, kind.toString());
-    String key = Keys.ArtifactCacheKey.createKey(type, name, location, "*");
-    return cacheUtils.getAllDataMatchingPattern(Keys.Kind.ARTIFACT.toString(), key).stream()
-        .sorted(
-            Comparator.comparing(
-                cd -> (String) cd.getAttributes().getOrDefault("creationTimestamp", "")))
-        .map(this::cacheDataToArtifact)
-        .filter(a -> account.equals(a.getMetadata("account")))
+    return accountResolver
+        .getCredentials(account)
+        .map(credentials -> credentials.list(kind, location).stream())
+        .orElseGet(Stream::empty)
```

Comment:
If we can't find the credentials, do we want to just return an empty list or throw some kind of error?

Reply:
Good question; I guess this is returning an empty list now.  (Also let me know if this mixing of Optional/Stream code is confusing...I'm not sure what i think of it.).

I picked this for consistency with the Manifest provider...but maybe we do want to throw an error, as presumably you can't reasonably pick a next version or decide what to delete if you don't have an account?

Though I also suspect that when I do a minor upcoming refactor I'll be able to pass in the credentials anyway so this might be moot.

Reply:
Yeah I was actually thinking of lifting the credentials resolution out of the ManifestProvider (into the controller, I guess) so it can fail as quickly as possible with an error appropriate to each endpoint, but we can leave this for now and see if we like that pattern any better 🤷‍♀️